debezium-core/src/main/java/io/debezium/relational/RelationalConnectorTask.java

```diff
@@ -0,0 +1,186 @@
+    private volatile RelationalDatabaseSchema schema;
+    private volatile Map<String, ?> lastOffset;
+
+    @Override
+    public void start(Configuration config) {
+        if (!state.compareAndSet(State.STOPPED, State.RUNNING)) {
+            LOGGER.info("Connector has already been started");
+            return;
+        }
+
+        StartingContext startingContext = getStartingContext(config);
+
+        RelationalDatabaseConnectorConfig connectorConfig = startingContext.getConnectorConfig();
+
+        Clock clock = Clock.system();
+
+        // Set up the task record queue ...
+        this.queue = new ChangeEventQueue.Builder<DataChangeEvent>()
+                .pollInterval(connectorConfig.getPollInterval())
+                .maxBatchSize(connectorConfig.getMaxBatchSize())
+                .maxQueueSize(connectorConfig.getMaxQueueSize())
+                .loggingContextSupplier(startingContext.getLoggingContextSupplier())
+                .build();
+
+        errorHandler = new ErrorHandler(startingContext.getConnectorClass(), connectorConfig.getLogicalName(), queue, this::cleanupResources);
+        TopicSelector<TableId> topicSelector = startingContext.getTopicSelector();
+
+        jdbcConnection = startingContext.getJdbcConnection();
+
+        this.schema = startingContext.getDatabaseSchema();
+
+        OffsetContext previousOffset = getPreviousOffset(startingContext.getOffsetContextLoader());
+        if (previousOffset != null && schema instanceof HistorizedRelationalDatabaseSchema) {
+            ((HistorizedRelationalDatabaseSchema) schema).recover(previousOffset);
+        }
+
+        EventDispatcher<TableId> dispatcher = new EventDispatcher<>(connectorConfig, topicSelector, schema, queue,
+                connectorConfig.getTableFilters().dataCollectionFilter(), DataChangeEvent::new);
+
+        coordinator = new ChangeEventSourceCoordinator(
+                previousOffset,
+                errorHandler,
+                startingContext.getConnectorClass(),
+                connectorConfig.getLogicalName(),
+                startingContext.getChangeEventSourceFactory(errorHandler, dispatcher, clock)
+        );
+
+        coordinator.start();
+    }
+
+    protected abstract StartingContext getStartingContext(Configuration config);
+
+    @Override
+    public List<SourceRecord> poll() throws InterruptedException {
+        List<DataChangeEvent> records = queue.poll();
+
+        List<SourceRecord> sourceRecords = records.stream()
+            .map(DataChangeEvent::getRecord)
+            .collect(Collectors.toList());
+
+        if (!sourceRecords.isEmpty()) {
+            this.lastOffset = sourceRecords.get(sourceRecords.size() - 1).sourceOffset();
+        }
+
+        return sourceRecords;
+    }
+
+    @Override
+    public void commit() throws InterruptedException {
+        if (lastOffset != null) {
+            coordinator.commitOffset(lastOffset);
+        }
+    }
+
```

Comment:
We should maybe implement a `commitRecord` method too that would as the default log the record into the log on `TRACE` level. It might help with problem solving.

Reply:
I'm not quite sure I'm following. Can you push a commit that does what you suggest?